example/docker-compose.yml

```diff
@@ -0,0 +1,76 @@
+version: '3.3'
+
+services:
+
+  filebeat:
+    image: docker.elastic.co/beats/filebeat:$ELK_VERSION
+    volumes:
+      ## Make sure to set this to the directory where your lancache logs are stored.
+      - ${CACHE_ROOT}/logs:/data/logs
+      - ./filebeat/filebeat.yml:/usr/share/filebeat/filebeat.yml
+    depends_on:
+      - logstash
+
+  ## ELK Stack
+  elasticsearch:
+    image: docker.elastic.co/elasticsearch/elasticsearch:$ELK_VERSION
+    volumes:
+      - type: bind
+        source: ./elasticsearch/elasticsearch.yml
+        target: /usr/share/elasticsearch/config/elasticsearch.yml
+        read_only: true
+      - type: volume
+        source: elasticsearch
+        target: /usr/share/elasticsearch/data
+    ports:
+      - "9200:9200"
+    environment:
+      ES_JAVA_OPTS: ${ES_JAVA_OPTS}
+      ELASTIC_PASSWORD: ${ELASTIC_PASSWORD}
+      discovery.type: single-node
+
+  logstash:
+    image: docker.elastic.co/logstash/logstash:$ELK_VERSION
+    volumes:
+      - type: bind
+        source: ./logstash/logstash.yml
+        target: /usr/share/logstash/config/logstash.yml
+        read_only: true
+      - type: bind
+        source: ../logstash.conf
+        target: /usr/share/logstash/pipeline/logstash.conf
+        read_only: true
+    environment:
+      LS_JAVA_OPTS: "-Xmx256m -Xms256m"
+      ELASTIC_PASSWORD: ${ELASTIC_PASSWORD}
+    depends_on:
+      - elasticsearch
+
+  kibana:
+    image: docker.elastic.co/kibana/kibana:$ELK_VERSION
+    volumes:
+      - type: bind
+        source: ./kibana/kibana.yml
+        target: /usr/share/kibana/config/kibana.yml
+        read_only: true
+    environment:
+      ELASTIC_PASSWORD: ${ELASTIC_PASSWORD}
+    ports:
+      - "5601:5601"
+    depends_on:
+      - elasticsearch
+  dashboard-importer:
+    image: appropriate/curl
+    environment:
+      ELASTIC_PASSWORD: ${ELASTIC_PASSWORD}
+    volumes:
+      - type: bind
+        source: ../everything.ndjson
+        target: /usr/share/tmp/everything.ndjson
+        read_only: true
+    command: sh -c "/usr/bin/curl -X POST http://kibana:5601/api/saved_objects/_import -u elastic:$ELASTIC_PASSWORD -H kbn-xsrf:true -F 'file=@/usr/share/tmp/everything.ndjson'"
```

Comment:
Two late night thoughts
1) this probably needs some forms of sleep to allow the kibana instance to actually come up
2) As written it will blat any customisations each time the stack comes up. needs a `[ -f /import_done ] || <curl> && touch /import_done` thing. (Late night scripting, my logic might be upside down, this should be tested :D

Reply:
@ewancolyer ^^

Reply:
In addition to sleeping, a `depends_on` statement could help.

Reply:
@GotenXiao it has a depends_on, but that'll be the container coming up, not the service actually initialising and being ready for api, which in my experience could be ~5minutes on a slow system with 1gb of memory!

Reply:
could use a `do while` loop instead and check the status code returning and keeping trying if it's not accepted it into the stack. Might be cleaner

Reply:
1. Will give this a test, I think the `/api/status` might work for us actually checking the readiness, rather than an arbitrary sleep value
2. I might see if I can have a env along the lines of `overwrite_dashboard` true/false which it checks for in the if statement to allow this to be easily changed.

Reply:
2. should be done, will probably finish off 1. tomorrow

Reply:
1. Has been done